src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs

```diff
@@ -287,7 +292,34 @@ private string ExtractGhesBaseUrl(string ghesApiUrl)
         }
     }
 
-    private async Task<(string GitArchiveUrl, string MetadataArchiveUrl, int GitArchiveId, int MetadataArchiveId)> GenerateArchive(
+    private async Task<string> UploadArchive(
+      string githubTargetOrg,
+      string awsBucketName,
+      bool useGithubStorage,
+      string archiveDownloadFilePath,
+      string archiveUploadFileName)
+    {
+#pragma warning disable IDE0063
+        await using (var archiveContent = _fileSystemProvider.OpenRead(archiveDownloadFilePath))
+#pragma warning restore IDE0063
+
+        if (useGithubStorage)
+        {
+            return await UploadArchiveToGithub(githubTargetOrg, archiveUploadFileName, archiveContent);
+        }
+#pragma warning disable IDE0046
+        else if (_awsApi.HasValue())
+#pragma warning restore IDE0046
+        {
+            return await UploadArchiveToAws(awsBucketName, archiveUploadFileName, archiveContent);
+        }
+        else
+        {
+            return await UploadArchiveToAzure(archiveUploadFileName, archiveContent);
+        }
```

Comment:
now that the upload method is uploading just one file I think we can switch to a `using declaration` syntax:
```suggestion
        await using var archiveContent = _fileSystemProvider.OpenRead(archiveDownloadFilePath);

        if (useGithubStorage)
        {
            return await UploadArchiveToGithub(githubTargetOrg, archiveUploadFileName, archiveContent);
        }

        if (_awsApi.HasValue())
        {
            return await UploadArchiveToAws(awsBucketName, archiveUploadFileName, archiveContent);
        }

        return await UploadArchiveToAzure(archiveUploadFileName, archiveContent);
```